config/config.go

```diff
@@ -20,8 +20,8 @@ var defaultAWSConfigResolvers = []awsConfigResolver{
 	resolveDefaultAWSConfig,
```

Comment:
So this is the order in which default config is loaded? If so, a comment would be helpful. 

Also how do we verify this is the order is consistent across SDKs

Reply:
This order is for how the sdk sources configuration from the user mostly then falls back to it's own default resolution behavior. E.g user provides custom http client then sdk augments that client with custom ca bundle.

But yeah docs would be helpful for us. This is internal behavior we're not exposing yet.  There isn't really a portable translation between SDKs for most of this.

Reply:
Though to call out this doesn't drive the default credentials configuration loading, as that is all handled by the `resolveCredentials` at the bottom of the list.